src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,54 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+		
+	public final String value;
+	private boolean isPrivate;
+	
+	/**
+	 * Validates given contact.
+	 * 
+	 * @param contact can be address, email or phone of person
+	 * @param isPrivate indicates if contact is private or not
+	 * @param validationRegex to check if given contact is valid
+	 * @param messageConstraints for user to handle case when given contact is invalid
+	 * @throws IllegalValueException
+	 */
+	public Contact(String contact, boolean isPrivate, String validationRegex, String messageConstraints) throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		if (!isValidContact(contact, validationRegex)) {
+			throw new IllegalValueException(messageConstraints);
```

Comment:
It's good that you use regex to do validation. But you need to recall what you learned about when/when not to use exceptions. This should be a part of input validation. And it is unusual to throw exceptions from a constructor.